util/uniq/slice.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package uniq provides dedulication utilities.
+package uniq
+
+import (
+	"reflect"
+)
+
+// Slice removes adjacent duplicate elements from the slice pointed to by sliceptr.
+// It zeros any indices removed from the tail.
+// eq reports whether the elements at i and j are equal.
+// It does O(len(*sliceptr)) operations.
+func Slice(sliceptr interface{}, eq func(i, j int) bool) {
+	rv := reflect.ValueOf(sliceptr).Elem()
```

Comment:
fail early and clearly if Value.Type().Kind() isn't Slice?

Reply:
Done